core/modules/translation/translation.block.inc

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * @file
+ * Extend the NodeBlock class to find the appropriate translation of a node,
+ * if available.
+ *
+ * This class overrides the prepare method of Block to replace the nid in the
+ * settings with the nid of the (node) translation of the content just before
+ * it is being rendered (primarily for the getTitle and getContent methods).
+ */
+
+class TranslatedNodeBlock extends NodeBlock {
+  protected $already_replaced = FALSE;
```

Comment:
I haven't tested but wondering if this should be a static variable.

Reply:
That makes sense though I'm trying to figure out how prepare will get called multiple times in order for already_replaced to be used. Maybe I need to try it out.

Reply:
I just looked with a debugger, and the whole already_replaced thing is unnecessary. I left it in since I first overrode getTitle and getContent and needed it.

Reply:
It's probably faster if we help you make a PR @sthomen so you can easily make the edits yourself. This time @klonos could update his PR but he'll need to know what it should be.

Reply:
Yeah, it seemed like things got stuck... I'll get myself up to speed on github.